### PR TITLE
Fixed an issue that would prevent attachments from being downloaded v…

### PR DESCRIPTION
**Pull request by javierquevedo**
…ia SSL connections when using a custom CA ceritficate that was included in the bundle.

Explanation:
All of the network operations use AFNetworking except for the download of files, which uses NSURLRequest. AFNetworking automatically checks which certificates are in the bundle and uses them to validate SSL connections. On the contrary, the MXMediaLoader class was relying only on those certificates that were manually accepted by the user. This could create a situation where all network connections worked, except for downloading media/attachments.
The MXMediaLoader now is capable of asking AFNetworking for those pinned certificates, and adds the CA to the chain of trust at the time of preparing the SSL negotiation.

### CODE REVIEW

File: MatrixSDK/Utils/Media/MXMediaLoader.m [hunk 1]
```diff
@@ -232,6 +233,14 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
     NSURLProtectionSpace *protectionSpace = [challenge protectionSpace];
     if ([protectionSpace.authenticationMethod isEqualToString:NSURLAuthenticationMethodServerTrust])
     {
+        NSSet <NSData *> *certificates = [AFSecurityPolicy certificatesInBundle:[NSBundle mainBundle]];
+        if (certificates && certificates.count > 0) {
```
Comment (manuroe) [inline MatrixSDK/Utils/Media/MXMediaLoader.m]: For code consistency, could you please add a new line before the opening bracket?

File: MatrixSDK/Utils/Media/MXMediaLoader.m [hunk 2]
```diff
@@ -232,6 +233,14 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
...
+            NSMutableArray *pinnedCertificates = [NSMutableArray array];
+            for (NSData *certificateData in certificates) {
```
Comment (manuroe) [inline MatrixSDK/Utils/Media/MXMediaLoader.m]: and here.

Comment (javierquevedo) [inline MatrixSDK/Utils/Media/MXMediaLoader.m]: Sure thing.

**Comment by manuroe** [comment]
oh, can you also change the base branch of this PR from "master" to "develop", please?
master contains only the code we have released.

**Comment by javierquevedo** [comment]
Signed-off-by: Javier Quevedojquevedo@gmail.com

**Comment by javierquevedo** [comment]
Signed-off-by: Javier Quevedo <jquevedo@gmail.com>

**Comment by javierquevedo** [comment]
Great. Thanks for accepting it. Next one will go to develop instead of master :)

**Pull request merged**